src/components/z-carousel/index.tsx

```diff
@@ -1,36 +1,284 @@
-import { Component, h, Prop } from "@stencil/core";
+import {
+  Component,
+  h,
+  Prop,
+  Element,
+  Watch,
+  Event,
+  EventEmitter,
+  State,
+  Host,
+} from "@stencil/core";
+import { HostElement } from "@stencil/core/internal";
+import {
+  CarouselArrowsPosition,
+  CarouselProgressMode,
+  ButtonVariantEnum,
+} from "../../beans";
 
 /**
  * ZCarousel component.
  * @cssprop --z-carousel-gutter - The gutter between items.
  * @slot - carousel items. use `<li>` elements inside this slot as it is wrapped inside an `<ul>`
  */
-
 @Component({
   tag: "z-carousel",
   styleUrl: "styles.css",
   shadow: false,
 })
 export class ZCarousel {
-  /** sets whether the z-carousel is on loading state */
-  @Prop() isLoading: boolean;
-  /** sets the height of z-carousel ghost loading, this prop is mandatory when isloading is set to true, as otherwise the component won't show. */
-  @Prop() ghostLoadingHeight: number = 100;
-  render() {
+  @Element() hostElement: HTMLZCarouselElement;
+
+  /** The z-carousel is on loading state */
+  @Prop()
+  isLoading: boolean;
+
+  /** The z-carousel title, if given. */
+  @Prop()
+  label?: string;
+
+  /** Shows only one content at a time */
+  @Prop()
+  single: boolean = false;
+
+  /** Arrow buttons position */
+  @Prop()
+  arrows: null | CarouselArrowsPosition;
```

Comment:
Questa la chiamerei `arrowsPosition`, come più sotto cambierei `progress` in `progressMode`

Reply:
corretto